src/components/Campaign.tsx

```diff
@@ -28,6 +31,28 @@ export const Campaign = () => {
   const campaignContentQuery = useGetCampaignContent(idCampaign);
   const campaignContentMutation = useUpdateCampaignContent();
 
+  const [showModal, hideModal] = useModal(() => {
+    const { campaignName, htmlContent, type, previewImage, design } =
+      campaignContentQuery.data!;
```

Comment:
In my opinion, the content is _the content_ and should not be restructured.

Reply:
campaignContentQuery return a Campaign while the Modal need a TemplateContent, both are differente

Reply:
We can resolve this  comment here https://github.com/FromDoppler/doppler-editors-webapp/pull/398#discussion_r1085406751

Reply:
Right, `<Campaign>` component should not know about templates. `<SaveAsTemplateModal>` component should create the template object.